models/models.js

```diff
@@ -62,4 +62,39 @@ exports.fetchArticles=()=>{
         })
         return rows;
     });
+}
+
+exports.fetchUserByUsername=(username)=>{
+
+    if(typeof username !== 'string'){
+        return Promise.reject({status:400,msg:'Invalid username'})
+    }
+
+    return db.query('SELECT * FROM users WHERE users.username=$1',[username]).then(({rows,rowCount})=>{
+        if(rowCount===0) {
+            return Promise.reject({status:404,msg:'Non-existent username'});
+        }
+        return rows
+    })
+}
+
+exports.addCommentByArticleId=(article_id,comment)=>{
+
+    //Did they say in lecture that pg-format sanitises input to protect against
+    //SQL injection?
```

Comment:
For literal values (%L) yes but not always. %s for example is a simple string and not escaped.